src/components/Box/__tests__/box-snapshot-tests.jest.js

```diff
@@ -0,0 +1,145 @@
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with roundedStart prop", () => {
+    const tree = renderer.create(<Box roundedStart={Box.roundedStarts.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with shadow prop", () => {
+    const tree = renderer.create(<Box shadow={Box.shadows.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with margin prop", () => {
+    const tree = renderer.create(<Box margin={Box.margins.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with marginX prop", () => {
+    const tree = renderer.create(<Box marginX={Box.marginXs.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with marginY prop", () => {
+    const tree = renderer.create(<Box marginY={Box.marginYs.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with marginTop prop", () => {
+    const tree = renderer.create(<Box marginTop={Box.marginTops.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with marginEnd prop", () => {
+    const tree = renderer.create(<Box marginEnd={Box.marginEnds.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with marginStart prop", () => {
+    const tree = renderer.create(<Box marginStart={Box.marginStarts.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with padding prop", () => {
+    const tree = renderer.create(<Box padding={Box.paddings.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with paddingX prop", () => {
+    const tree = renderer.create(<Box paddingX={Box.paddingXs.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with paddingY prop", () => {
+    const tree = renderer.create(<Box paddingY={Box.paddingYs.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with paddingTop prop", () => {
+    const tree = renderer.create(<Box paddingTop={Box.paddingTops.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with paddingEnd prop", () => {
+    const tree = renderer.create(<Box paddingEnd={Box.paddingEnds.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with paddingBottom prop", () => {
+    const tree = renderer.create(<Box paddingBottom={Box.paddingBottoms.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with paddingStart prop", () => {
+    const tree = renderer.create(<Box paddingStart={Box.paddingStarts.SMALL} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with backgronudColor prop", () => {
+    const tree = renderer.create(<Box BackgronudColor={Box.backgroundColors.PRIMARY_BACKGROUND_COLOR} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("with text prop", () => {
+    const tree = renderer.create(<Box textColor={Box.textColors.PRIMARY_COLOR} />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
```

Comment:
Super important: let's add a test for "with children prop."
Without children prop support this component will be broken.